streams/src/main/java/org/apache/kafka/streams/processor/internals/ProcessorStateManager.java

```diff
@@ -381,6 +356,31 @@ public void close(Map<TopicPartition, Long> ackedOffsets) throws IOException {
         }
     }
 
+    // write the checkpoint
+    @Override
+    public void checkpoint(final Map<TopicPartition, Long> ackedOffsets) {
```

Comment:
Nah, the offsets are acked first. They need to be otherwise we risk writing a checkpoint for data that has not been acked.